src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/JWKKeyAccessor.kt

```diff
@@ -0,0 +1,78 @@
+import java.security.spec.RSAPublicKeySpec
+
+@Component
+class JWKKeyAccessor(
+  @Value("\${jwt.signing.key.pair}") private val privateKeyPair: String,
+  @Value("\${jwt.keystore.password}") private val keystorePassword: String,
+  @Value("\${jwt.keystore.alias:elite2api}") private val keystoreAlias: String,
+  @Value("\${jwt.jwk.key.id}") private val keyId: String,
+  @Value("#{'\${jwt.auxiliary.keystore.alias}' <= '' ? null : '\${jwt.auxiliary.keystore.alias}'}") private val keystoreAliasAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.keystore.password}' <= '' ? null : '\${jwt.auxiliary.keystore.password}'}") private val keystorePasswordAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.signing.key.pair}' <= '' ? null : '\${jwt.auxiliary.signing.key.pair}'}") private val privateKeyPairAuxiliary: String?,
+  @Value("#{'\${jwt.auxiliary.jwk.key.id}' <= '' ? null : '\${jwt.auxiliary.jwk.key.id}'}") private val keyIdAuxiliary: String?,
+) {
+  private val primaryKeyStore: KeyStore = initializePrimaryKeyStore()
+  private val auxiliaryKeyStore: KeyStore = initializeAuxiliaryKeyStore()
+  fun getPrimaryKeyPair() = getKeyPair(primaryKeyStore, keystoreAlias, keystorePassword)
+
+  fun getAuxiliaryKeyPair(): KeyPair? {
+    return if (ObjectUtils.anyNull(keystoreAliasAuxiliary, keystorePasswordAuxiliary, keyIdAuxiliary)) {
+      null
+    } else {
+      return getKeyPair(auxiliaryKeyStore, keystoreAliasAuxiliary, keystorePasswordAuxiliary)
+    }
+  }
+
+  private fun getKeyPair(keyStore: KeyStore, keystoreAlias: String?, keystorePassword: String?): KeyPair {
+    val privateKey =
+      keyStore.getKey(keystoreAlias, keystorePassword!!.toCharArray()) as RSAPrivateCrtKey
+    val rsaPublicKeySpec = RSAPublicKeySpec(privateKey.modulus, privateKey.publicExponent)
+    val publicKey = KeyFactory.getInstance("RSA").generatePublic(rsaPublicKeySpec)
+    return KeyPair(publicKey, privateKey)
+  }
+  fun getAuxiliaryPublicKey(): RSAKey? {
+    return if (ObjectUtils.anyNull(keyIdAuxiliary, keystoreAliasAuxiliary, keystorePasswordAuxiliary, privateKeyPairAuxiliary)) {
+      null
+    } else {
+      return buildRSAKey(getAuxiliaryKeyPair(), keyIdAuxiliary)
+    }
+  }
+
+  fun getPrimaryPublicKey() = buildRSAKey(getPrimaryKeyPair(), keyId)
+
+  private fun initializeKeyStore(privateKeyPair: String?, keystorePassword: String?): KeyStore {
+    val store = KeyStore.getInstance("jks")
+    val auxiliaryKeyStoreInputStream: InputStream? = privateKeyPair?.let { ByteArrayResource(decodeBase64(it)) }?.inputStream
+
+    auxiliaryKeyStoreInputStream.use {
+      store.load(it, keystorePassword?.toCharArray())
+    }
+    return store
+  }
+  private fun initializePrimaryKeyStore() = initializeKeyStore(privateKeyPair, keystorePassword)
+
+  private fun initializeAuxiliaryKeyStore() = initializeKeyStore(privateKeyPairAuxiliary, keystorePasswordAuxiliary)
+
+  private fun buildRSAKey(keyPair: KeyPair?, keyId: String?): RSAKey {
```

Comment:
This doesn't need to be nullable, when passing in the getAuxiliaryKeyPair(), use the `!!` operator

Reply:
In this case enough null checks are there , change to !! operator